ITEP-68793: Support Volumetric Analytics in Scene Controller (#148)
open-edge-platform/scenescape: docs/user-guide/How-to-define-object-properties.md

# How to Define Object Properties

Object Library allows you to configure various properties for object categories in SceneScape. This guide walks through the process of defining and customizing object properties.

## Working with the Object Library

1. Navigate to the SceneScape homepage.
2. Click on the Object tab from the topbar.

### Add a New Object

1. Click on "New Object".
2. Input the object properties.
3. Click on "Add New Object".

![Create New Object](images/new-object.png)

### Update Existing Object

> 1. Click on the "Spanner" icon in the Update column next to the object to be edited.

Reviewer: "Spanner" is British English, so Americans may not comprehend it. I think there is a UTF-8 wrench/spanner icon, so maybe we could use that? It looks pretty close to the Bootstrap icon wrench.

```suggestion
1. Click on the wrench/spanner (🔧) icon in the Update column in the row of the object to be edited.
```

2. Edit the object properties.

> 3. Click on "Update Object".

Reviewer: ```suggestion
3. Click on "Update Object" to save any changes.
```

## Basic Object Properties

### Size Configuration

- **Object size in x-axis**: Define the width of the object in meters
- **Object size in y-axis**: Define the length of the object in meters
- **Object size in z-axis**: Define the height of the object in meters

### Tracking Behavior Settings

> - **Matching Threshold Radius**: Set the maximum distance for matching object tracks

Reviewer: In the UI, the label is "Tracking radius (meters)", so we should probably use this here.

```suggestion
- **Tracking radius (meters)**: Set the maximum distance from the object center for matching new detections with the track
```

> - **Object Shift Type**: Shift type is used to compute the bottom center of the object to estimate its position in world coordinates.

Reviewer: Similar to the above, since the UI label isn't "Object Shift Type":

```suggestion
- **Shift type**: Shift type is used to compute the bottom center of the object to estimate its position in world coordinates.
```

  - For most objects the default setting of "Type 1" will work well.
  - For wide and short objects, "Type 2" performs better.

## Additional Settings

> - **Infer Rotation from Velocity**: When enabled, orientation of the object is inferred from the computed velocity.

Reviewer: ```suggestion
- **Rotation from velocity**: When enabled, orientation of the object is inferred from the computed velocity.
```

> - **Project to Map**: When enabled, objects will be projected onto the map surface.

Reviewer: ```suggestion
- **Project to map**: When enabled, objects will be projected onto the map surface.
```

## 3D Model Configuration

> By default, the shape of the object is a cuboid. Instead, the user can provide a 3D asset file (glb) for 3D visualization.

Reviewer: ```suggestion
By default, the shape of the object is a cuboid. Instead, the user can provide a 3D asset file (.glb) for 3D visualization.
```

> 1. Click on "New Object" or Click on the "Spanner" icon in the Update column next to the object to be edited.

Reviewer: ```suggestion
1. Click on "New Object" or click on the wrench/spanner (🔧) icon in the Update column in the row of the object to be edited.
```

> 2. Choose a glb file with the file picker input.

Reviewer: ```suggestion
2. Choose a .glb file with the file picker input.
```

3. Edit any of the asset property fields.
4. Click on "Add New Object" or "Update Object"

### Asset Properties

1. **Scale Adjustment**:
   - Use the scale value to resize the model uniformly along X, Y, and Z axes.

2. **Orientation Adjustment**:
   - Rotate the 3D asset along X, Y, and Z axes to set the default orientation.

3. **Position Adjustment**:
   - Adjust the default position of the 3D asset wrt origin.

![Add GLB as Object Asset](images/object-glb.png)

### Verify Results

1. Navigate to the 3D UI of a Scene.

> 2. Instead of the default cuboid, the uploaded GLB will represent the tracked object.

Reviewer: ```suggestion
2. Instead of the default cuboid, the uploaded 3D asset (.glb) will represent the tracked object.
```
